app/services/xccdf_report_parser.rb

```diff
@@ -30,6 +30,11 @@ def check_report_format
     )
   end
 
+  def valid_message_format?(message)
+    message['id'].present? && message['metadata'].present? &&
+      message['metadata']['fqdn'].present?
```

Comment:
This is a good case for `dig`:

```suggestion
    message['id'].present? && message.dig('metadata', 'fqdn').present?
```